src/Schemes/Online_r2.jl

```diff
@@ -377,13 +377,16 @@ function checkpoint_struct_while(
             for j= oldcapo:(next_action.iteration-1)
                 if go
                     body(model)
-                    go = condition(model)
+                    onlinesteps=onlinesteps+1
+                else
+                    break
                 end
-                onlinesteps=onlinesteps+1
+                go = condition(model)
```

Comment:
The logic here is hard to follow. The assigment`go = condition(model)` can just be placed as the first statement of the loop body and the repeated statement inside the if-true branch can be removed.